doc/articles/concepts/source-generation.md

```diff
@@ -0,0 +1,42 @@
+# Source Generation Overview
+
```

Comment:
This is an overview document and thus no need to drown folks in huge amount of details. Let's give them the base concepts that they need to be aware of and link out to some breadcrumbs if they want to go deeper.